### DISCUSSION TITLE
SLL Not Working

### DISCUSSION THREAD

**Discussion by ghost** [inspircd/inspircd#2094] [Support]
Hello,

I followed this tutorial https://www.linode.com/docs/guides/install-and-configure-inspircd-on-debian-10-ubuntu-2004/ but can't get it to work with SSL, it works withtout SLL but with SSL I get protocol error

**Comment by sadiepowell** [comment]
That tutorial is extremely outdated. You need to:

1. Load the ssl_gnutls (assuming you're using the Debian v3 package) with `<module name="ssl_gnutls">`.
2. Add a bind tag which specifies a SSL (TLS) profile like [this](https://docs.inspircd.org/3/modules/ssl_gnutls/#bind).
3. Add a [SSL profile](https://docs.inspircd.org/3/modules/ssl_gnutls/#sslprofile) with the same name as used in your `<bind>` tag.
4. Restart your server.

InspIRCd ships with a tool called `inspircd-testssl` which you can run against your server to test your SSL (TLS) configuration.

You may also want to load [the sslrehashsignal module](https://docs.inspircd.org/3/modules/sslrehashsignal/) which allows you to reload your SSL (TLS) certificates in a post-deploy hook by sending it the SIGUSR1 signal.

**Reply by ghost** [reply]
Thanks! If I have only the cert.pem and key.pem can I leave the:  `cafile="" ; crlfile="" ; dhfile=""`  blank?

**Reply by sadiepowell** [reply]
Generally yes, but on GnuTLS versions older than 3.6.0 you may be required to provide a DH parameters file. You can generate one with `certtool --generate-dh-params --sec-param normal --outfile dhparams.pem`.

**Reply by ghost** [reply]
I will try tomorrow, got the limit of certificates generated :/

**Comment by ghost** [comment]
Everything working now! You can close this. thanks!
